chain/client/src/client.rs

```diff
@@ -1662,197 +1662,3 @@ impl Client {
         Ok(())
     }
 }
-
-#[cfg(test)]
-mod test {
-    use std::collections::HashMap;
-    use std::path::Path;
-    use std::sync::Arc;
-
-    use cached::Cached;
-
-    use near_chain::Provenance;
-    use near_chain::{ChainGenesis, RuntimeAdapter};
-    use near_chain_configs::Genesis;
-    use near_chunks::test_utils::ChunkForwardingTestFixture;
-    use near_chunks::ProcessPartialEncodedChunkResult;
-    use near_crypto::KeyType;
-    use near_primitives::block::{Approval, ApprovalInner};
-    use near_primitives::hash::hash;
-    use near_primitives::validator_signer::InMemoryValidatorSigner;
-    use near_primitives::version::ProtocolFeature;
-    use near_primitives::version::PROTOCOL_VERSION;
-    use near_store::test_utils::create_test_store;
-    use nearcore::config::GenesisExt;
```

Comment:
Hm, so this is actually the only import which needs dependency on `nearcore` -- to get a `GenesisExt`. The rest of the code here works without `nearcore` just fine. 

I wonder if the problem here is that `GenesisExt` is a part of `nearcore`? 

My understanding is that:

* The `Genesis` struct is defined in `near-chain-configs`, which is one of the leaf crates (it only depends on the primitives).
* The `GenesisExt` is defined in `nearcore`, which is *the* root crate, it depends on everything. At the same time, `GenesisExt` just adds a bunch of additional constructors to `Genesis`, it doesn't seem to fundamentally need nearcore.


So, I am wondering, can we just define the `GenesisExt` in `near-chain-configs` instead? @bowenwang1996 is there some reason why that might *not* work?

Reply:
That is, I am wondering if we should explore the following alternative:

* leave these tests as they are
* move genesis ext to the near-chain-config
* remove dependency on `nearcore` without moving the tests.

Reply:
Let me look into it.

Reply:
While combining `GenesisExt` and `Genesis` in the same module make sense and should be done eventually, `near-client` also depends on `nearcore::NightshadeRuntime`: https://github.com/near/nearcore/blob/master/chain/client/src/client.rs#L1701

Reply:
Ah, right, this means that we have move these tests!